modules/packages/manifests/telegraf.pp

```diff
@@ -2,9 +2,17 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-class packages::telegraf {
+class packages::telegraf (
+    Pattern[/^\d+\.\d+\.\d+_?\d*$/] $version = '1.12.3',
+) {
+    # arm64 if Apple processor
+    if /^Apple.*/ in $facts['processors']['models'] {
+        $suffix = '-arm64'
+    } else {
+        $suffix = ''
+    }
```

Comment:
Another way to test for M1 is `model_identifier => "Macmini9,1",` if you want to keep it consistent with existing methods.  Not a big deal, just a nit.

Reply:
I understand matching previous checking of macmini model. For this I like that the condition shows that we are looking for the processor architecture and *could work with any apple silicon model.

Reply:
That is fine.  FYI, "Macmini9,1" models are only M1 arm64 arch.  There are no x86_64 models and there never will be.